sklearn/lda.py

```diff
@@ -450,13 +450,13 @@ def transform(self, X):
         X_new : array, shape (n_samples, n_components)
             Transformed data.
         """
-        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
-
-        X = check_array(X)
         if self.solver == 'lsqr':
             raise NotImplementedError("transform not implemented for 'lsqr' "
                                       "solver (use 'svd' or 'eigen').")
-        elif self.solver == 'svd':
+        check_is_fitted(self, ['xbar_', 'scalings_'], all_or_any=any)
```

Comment:
Well my concern was that `check_is_fitted` is fine with only `xbar_` fitted which is useful nowhere. So I would remove `xbar_` at least from that check for logical consistency. 

For completeness a `check_is_fitted(self, 'xbar_')` could then go in the appropriate decision branch. But I don't insist on that one because from what I see `check_is_fitted` is about throwing graceful errors if the variable concerned hasn't been fitted. For `x_bar_` not to be fitted if `solver == 'svd'`, the user must have called fit with a different solver and then changed the solver by hand, which I consider advanced (and weird), so that the user can then also deal with the ensuing error message.

This basically also invalidates what I said above. I will post this message to be able to remember the reasoning, but I agree with you that nothing needs to be changed.